injected/src/features/autofill-password-import.js

```diff
@@ -0,0 +1,251 @@
+        return await withExponentialBackoff(fn)
+    }
+
+    /**
+     * @returns {Promise<HTMLElement|Element|null>}
+     */
+    async findSignInButton () {
+        return await withExponentialBackoff(() => document.querySelector(this.signinButtonSelector))
+    }
+
+    /**
+     * Checks if the path is supported and animates/taps the element if it is.
+     * @param {string} path
+     */
+    async handleElementForPath (path) {
+        // FIXME: we need to check if the path is supported, otherwise
+        // for some reason google doesn't wait to proceed with the signin step.
+        // Not too sure why this is happening, there are no errors on the console.
+
+        if ([this.#exportButtonSettings?.path, this.#settingsButtonSettings?.path, this.#signInButtonSettings?.path].indexOf(path) !== -1) {
+            try {
+                const { element, style, shouldTap } = await this.getElementAndStyleFromPath(path) ?? {}
+                if (element != null) {
+                    shouldTap ? this.autotapElement(element) : this.animateElement(element, style)
+                }
+            } catch {
+                console.error('password-import: handleElementForPath failed for path:', path)
+            }
+        }
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonContainerSelector () {
+        return this.#exportButtonSettings?.selectors?.join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonLabelTextSelector () {
+        return this.#exportButtonSettings?.labelTexts
+            .map(text => `button[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinLabelTextSelector () {
+        return this.#signInButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]:not([target="_top"])`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinButtonSelector () {
+        return `${this.#signInButtonSettings?.selectors?.join(',')}, ${this.signinLabelTextSelector}`
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsLabelTextSelector () {
+        return this.#settingsButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsButtonSelector () {
+        return `${this.#settingsButtonSettings?.selectors?.join(',')}, ${this.settingsLabelTextSelector}`
+    }
+
+    setButtonSettings () {
+        this.#exportButtonSettings = this.getFeatureSetting('exportButton')
+        this.#signInButtonSettings = this.getFeatureSetting('signInButton')
+        this.#settingsButtonSettings = this.getFeatureSetting('settingsButton')
+    }
+
+    init () {
+        this.setButtonSettings()
+
+        const handleElementForPath = this.handleElementForPath.bind(this)
+        const historyMethodProxy = new DDGProxy(this, History.prototype, 'pushState', {
```

Comment:
Did we do this already for element hiding? We should probably share that same code eventually. Can you make it a follow up (I'm aware they currently won't run at the same time)